src/common/stream_coder.h

```diff
@@ -0,0 +1,23 @@
+#ifndef STREAM_CODER_H
+#define STREAM_CODER_H
+
+#include "allocator.h"
+#include <nan.h>
+
+using namespace v8;
+
+class StreamCoder : public Nan::ObjectWrap {
+  public:
+    const uint8_t* next_in;
+    size_t available_in;
```

Comment:
Sorry (I guess the language barrier says hi…) – there’s a `BrotliEncoderCompressStream()` method declared in `encode.h` that works pretty much like `BrotliDecompressStream()`, so using that might be nice to avoid code duplication.

I’m not sure how that interacts with “filling the blockSize” the way it’s currently done in `index.js`, though.